nutils/function.py

```diff
@@ -3789,6 +3789,15 @@ def __init__(self, *, default_geometry_name='x'):
     super().__setattr__('default_geometry_name', default_geometry_name)
     super().__init__()
 
+  def __getstate__(self):
+    'Pickle instructions'
+    attrs = '_arg_shapes', '_attributes', 'default_geometry_name'
+    return {k: getattr(self, k) for k in attrs}
+
+  def __setstate__(self, d):
+    'Unpickle instructions'
+    for k, v in d.items(): super().__setattr__(k, v)
+
```

Comment:
Since pickle bypasses `__init__`  when unpickling an object with `__getstate__`, `__setstate__`, the attribute `arg_shapes` is undefined for unpickled `Namespace` objects. Could you fix this by converting the `arg_shapes` attribute into a property?